### ISSUE TITLE
Extracting parts of TracePage into separate package.

### ISSUE THREAD

**Issue by aocenas** [jaegertracing/jaeger-ui#507]
## Requirement - what kind of business use case are you trying to solve?
At Grafana we would like to integrate Jaeger UI, specifically the TraceGraph and TraceTimelineViewer components, into Grafana.

## Problem - what in Jaeger blocks you from solving the requirement?
Jaeger already has the ability to provide [embeddable view of the trace](https://www.jaegertracing.io/docs/next-release/frontend-ui/#trace-page). For our needs that is not usable due to requirements:
- Themeability: We want to make it as close in look and feel to rest of Grafana as possible.
- Pluggable behaviour: Things like custom actions, additional buttons or elements.
- Support for other backends: Allow displaying trace from any kind of backend (data transforms is not meant as part of this issue) and not depend on Jaeger server.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

At this moment it seems that the work can be split to 3 semi separate parts:

- Extract the TracePage component or its subcomponents into separate package while removing global dependencies like redux, url state or tracking.
- Make the resulting component(s) themeable.
- Add pluggability for custom behaviour.

There are more details in this document: https://docs.google.com/document/d/1fRpiNdj3NmxDGpvkW49MlI19M5RdPa2GZloDsaz8PxU/edit?usp=sharing

As this is a bit bigger scope I would like to track the high level concerns and comments here and create separate issues for each step to be able to discuss each in more details without making the discussion confusing.

**Comment by aocenas** [comment]
Hey @everett980 @rubenvp8510, hope splitting this into separate issues makes sense as this could be considered an epic rather than single task.

**Comment by everett980** [comment]
I agree that splitting this into separate tasks makes sense and will help with separating discussions.

Though I foresee #509 and #510 both possibly being very dependent on #508 so conversations can't be completely isolated.
